Yafc.Model.Tests/LuaDependentTestHelper.cs

```diff
@@ -35,19 +35,23 @@ static LuaDependentTestHelper() {
     internal static Project GetProjectForLua(string targetStreamName = null) {
         // Verify correct non-parallel declaration for tests, to accomodate the singleton analyses.
         StackTrace stack = new();
+
         for (int i = 1; i < stack.FrameCount; i++) {
             // Search up the stack until we find a method with [Fact] or [Theory].
             MethodBase method = stack.GetFrame(i).GetMethod();
+
             if (method.GetCustomAttribute<FactAttribute>() != null || method.GetCustomAttribute<TheoryAttribute>() != null) {
                 targetStreamName ??= method.DeclaringType.FullName + '.' + method.Name + ".lua";
 
                 // CollectionAttribute doesn't store its constructor argument, so we have to read the attribute data instead of the constructed attribute.
                 CustomAttributeData data = method.DeclaringType.GetCustomAttributesData().SingleOrDefault(d => d.AttributeType == typeof(CollectionAttribute), false);
+
                 if ((string)data?.ConstructorArguments[0].Value != "LuaDependentTests") {
```

Comment:
To give an example where i think the code block shouldnt be split right before the if:
Here the `method` variable i made specifically for the `if`, and it would make more sense to have a new line after the for, but no newline before the if. Then again the same thing with `data` below.

The entire code however is long and dense enough, especially with all the comments in-between, that a few newlines do improve things regardless of where you place them.

```suggestion
        for (int i = 1; i < stack.FrameCount; i++) {

            // Search up the stack until we find a method with [Fact] or [Theory].
            MethodBase method = stack.GetFrame(i).GetMethod();
            if (method.GetCustomAttribute<FactAttribute>() != null || method.GetCustomAttribute<TheoryAttribute>() != null) {

                targetStreamName ??= method.DeclaringType.FullName + '.' + method.Name + ".lua";

                // CollectionAttribute doesn't store its constructor argument, so we have to read the attribute data instead of the constructed attribute.
                CustomAttributeData data = method.DeclaringType.GetCustomAttributesData().SingleOrDefault(d => d.AttributeType == typeof(CollectionAttribute), false);
                if ((string)data?.ConstructorArguments[0].Value != "LuaDependentTests") {
```